### ISSUE TITLE
Could not connect to server - kubernetes-cockpit running as a POD

### ISSUE THREAD

**Issue by VinayKumarbrsoft** [cockpit-project/cockpit#4790]
I am running cockpit as pod in my local cluster and when I try to access cockpit GUI I get following error:

```
Couldn't connect to server
Couldn't find running API server
```

The pod logs gives following error:

```
MESSAGE: cockpit-protocol: http://localhost:8080/api: couldn't connect: Connection refused
MESSAGE: cockpit-protocol: https://localhost:8443/api: couldn't connect: Connection refused
MESSAGE: cockpit-protocol: https://localhost:6443/api: couldn't connect: Connection refused

```
Could you please point me in right direction what might be wrong?  Kubernetes cluster is running cockpit as pod and the POD is in running status.

**Comment by VinayKumarbrsoft** [comment]
You are right!! I am using cockpit/kubernetes container.  Its bit strange, I am able to talk to my Kubernetes cluster using the same user credentials via kubectl.   If I recreate my container I am able to login fine then after sometime cockpit just refuses to retrieve information.  Have you seen similar problem?

**Comment by petervo** [comment]
Can we see any console output as well as the the login-data in session storage?

**Comment by VinayKumarbrsoft** [comment]
`window.sessionStorage.getItem('login-data')
"{"apiVersion":"v1","clusters":[{"cluster":{"certificate-authority-data":"LS0tLS1CRUdJTiBDRV","insecure-skip-tls-verify":false,"server":"https://100.0.0.1:443"},"name":"container-cluster"}],"contexts":[{"context":{"cluster":"container-cluster","user":"test"},"name":"container-context"}],"current-context":"container-context","displayName":"test","users":[{"name":"test","user":{"password":"changeme","username":"test"}}]}"`

There are too many of the below errors in my java script console .  

`Object {options: Object, statusText: "cancelled", problem: "cancelled", status: 999}
options
:
Object
problem
:
"cancelled"
status
:
999
statusText
:
"cancelled"
__proto__
:
Object
__defineGetter__
:
__defineGetter__()
__defineSetter__
:
__defineSetter__()
__lookupGetter__
:
__lookupGetter__()
__lookupSetter__
:
__lookupSetter__()
constructor
:
Object()
hasOwnProperty
:
hasOwnProperty()
isPrototypeOf
:
isPrototypeOf()
propertyIsEnumerable
:
propertyIsEnumerable()
toLocaleString
:
toLocaleString()
toString
:
toString()
valueOf
:
valueOf()
get __proto__
:
__proto__()
set __proto__
:
__proto__()`

**Comment by petervo** [comment]
Interesting seems that the correct auth attempt is dropped. If you clear your console and then hit the refresh, does it still have that "transport closed, dropped message" line?

I'm assuing the tls":{"authority":{"data":" entry is just truncated and what's should there is not the actual value.

**Comment by VinayKumarbrsoft** [comment]
How  many users can logon to cockpit using the same user credentials? We are using the same credentials to access the cluster for different reasons.   Do we need to have dedicated user created just for cockpit?

**Comment by petervo** [comment]
No you shouldn't need a separate user for cockpit. There shouldn't be a limit on how many users can log in.

If you are disconnected like this automatically does the reconnect button work? Or does it only work again if you log out and back in?

What version of kubernetes are you running?

**Comment by VinayKumarbrsoft** [comment]
@petervo Sorry for not replying to you earlier.   If I am disconnected automatically the reconnect button does not work.  I need to logout/sometimes it automatically logs out itself.  I am running 1.2 version of Kubernetes.  @stefwalter seems to have already fixed the problem.  Did you guys managed to replicate the problem?

**Comment by petervo** [comment]
I closed this with a possible fix in #4800. If you see the problem after the next release (117 or greater) please reopen.
